### PR TITLE
enhancement(subnet): added support for ibm_is_subnet_routing_table_attachment

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
--- PASS: TestAccIBMISSubnetRoutingTableAttachment_basic (125.18s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm 129.326s
```

### CODE REVIEW

File: ibm/resource_ibm_is_subnet_routing_table_attachment.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+	replaceSubnetRoutingTableOptionsModel := new(vpcv1.ReplaceSubnetRoutingTableOptions)
+	replaceSubnetRoutingTableOptionsModel.ID = &subnet
+	replaceSubnetRoutingTableOptionsModel.RoutingTableIdentity = routingTableIdentityModel
+	resultACL, response, err := sess.ReplaceSubnetRoutingTable(replaceSubnetRoutingTableOptionsModel)
+
+	if err != nil {
+		log.Printf("[DEBUG] Error while attaching a routing table to a subnet %s\n%s", err, response)
+		return fmt.Errorf("Error while attaching a routing table to a subnet %s\n%s", err, response)
+	}
+	d.SetId(subnet)
```
Comment (deepaksibm) [inline ibm/resource_ibm_is_subnet_routing_table_attachment.go:152]: should it be a composite ID like <subnet_id>/<routingtable_id> ?

File: ibm/resource_ibm_is_subnet_routing_table_attachment.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+	// Construct an instance of the RoutingTableIdentityByID model
+	routingTableIdentityModel := new(vpcv1.RoutingTableIdentityByID)
+	routingTableIdentityModel.ID = &routingTable
+
+	// Construct an instance of the ReplaceSubnetRoutingTableOptions model
+	replaceSubnetRoutingTableOptionsModel := new(vpcv1.ReplaceSubnetRoutingTableOptions)
+	replaceSubnetRoutingTableOptionsModel.ID = &subnet
+	replaceSubnetRoutingTableOptionsModel.RoutingTableIdentity = routingTableIdentityModel
+	resultACL, response, err := sess.ReplaceSubnetRoutingTable(replaceSubnetRoutingTableOptionsModel)
```
Comment (deepaksibm) [inline ibm/resource_ibm_is_subnet_routing_table_attachment.go:146]: resultRoutingTable or just routingTable

File: ibm/resource_ibm_is_subnet_routing_table_attachment.go [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+	id := d.Id()
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	getSubnetRoutingTableOptionsModel := &vpcv1.GetSubnetRoutingTableOptions{
+		ID: &id,
+	}
+	nwacl, response, err := sess.GetSubnetRoutingTable(getSubnetRoutingTableOptionsModel)
```
Comment (deepaksibm) [inline ibm/resource_ibm_is_subnet_routing_table_attachment.go:169]: nwacl -> routingTable

File: ibm/resource_ibm_is_subnet_routing_table_attachment.go [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+		log.Printf("[DEBUG] vpc default routing table is not null :%s", *vpc.DefaultRoutingTable.ID)
+		// Construct an instance of the RoutingTableIdentityByID model
+		routingTableIdentityModel := new(vpcv1.RoutingTableIdentityByID)
+		routingTableIdentityModel.ID = vpc.DefaultRoutingTable.ID
+
+		// Construct an instance of the ReplaceSubnetRoutingTableOptions model
+		replaceSubnetRoutingTableOptionsModel := new(vpcv1.ReplaceSubnetRoutingTableOptions)
+		replaceSubnetRoutingTableOptionsModel.ID = &id
+		replaceSubnetRoutingTableOptionsModel.RoutingTableIdentity = routingTableIdentityModel
+		resultACL, response, err := sess.ReplaceSubnetRoutingTable(replaceSubnetRoutingTableOptionsModel)
```
Comment (deepaksibm) [inline ibm/resource_ibm_is_subnet_routing_table_attachment.go:284]: modify all ACL references

File: website/docs/r/is_subnet_routing_table_attachment.html.markdown [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  region = "eu-gb"
+}
+```
+
+## Example usage
+
+```terraform
+  resource "ibm_is_vpc" "example" {
+		name = "example-vpc"
+	}
```
Comment (deepaksibm) [inline website/docs/r/is_subnet_routing_table_attachment.html.markdown:29]: fix indentation

File: website/docs/r/is_subnet_routing_table_attachment.html.markdown [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+		depends_on    = [ibm_is_vpc_routing_table.example, ibm_is_subnet.example]
+		subnet        = ibm_is_subnet.example.id
+		routing_table = ibm_is_vpc_routing_table.example.routing_table
+  }
+
+```
+## Argument reference
+Review the argument references that you can specify for your resource. 
+
+- `routing_table` - (Required, Forces new resource, String) The routing table identity.
```
Comment (deepaksibm) [inline website/docs/r/is_subnet_routing_table_attachment.html.markdown:52]: routing table id is patchable so can we remove forces new resource

File: website/docs/r/is_subnet_routing_table_attachment.html.markdown [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+	resource "ibm_is_subnet" "example" {
+		name                        = "example-subnet"
+		vpc                         = ibm_is_vpc.example.id
+		zone                        = "eu-gb-1"
+		total_ipv4_address_count    = 16
+	}
+
+	resource "ibm_is_subnet_routing_table_attachment" "example" {
+		depends_on    = [ibm_is_vpc_routing_table.example, ibm_is_subnet.example]
```
Comment (deepaksibm) [inline website/docs/r/is_subnet_routing_table_attachment.html.markdown:43]: can we remove depends on as we are already referring routing table and subnet resources in the required arguments

File: ibm/resource_ibm_is_subnet_routing_table_attachment.go [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+}
+
+func resourceIBMISSubnetRoutingTableAttachmentRead(d *schema.ResourceData, meta interface{}) error {
+	id := d.Id()
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	getSubnetRoutingTableOptionsModel := &vpcv1.GetSubnetRoutingTableOptions{
```
Comment (deepaksibm) [inline ibm/resource_ibm_is_subnet_routing_table_attachment.go:166]: if we make a composite ID, we can split the id and use subnet ID here to support the terraform import

File: ibm/resource_ibm_is_subnet_routing_table_attachment.go (hunk 1 above)
Comment (ujjwal-ibm) [reply ibm/resource_ibm_is_subnet_routing_table_attachment.go:152]: not needed since subnet id is enough

**Comment by ujjwal-ibm** [comment]
moving to upstream

**Pull request closed without merging**